### PR TITLE
Adding polling to all node readiness

### PR DESCRIPTION
**Pull request by krousey**
#10622

cc @quinton-hoole

### CODE REVIEW

**Comment by krousey** [comment]
From the current PR builder running right now:

```
INFO: Waiting up to 1m0s for all nodes to be ready
INFO: Node pull-e2e-3-minion-pwnk condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transistion time: 2015-07-09 11:47:34 -0700 PDT
INFO: Successfully found node pull-e2e-3-minion-pwnk readiness to be true
INFO: Node pull-e2e-3-minion-zlm9 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transistion time: 2015-07-09 11:47:34 -0700 PDT
INFO: Successfully found node pull-e2e-3-minion-zlm9 readiness to be true
```

**Comment by brendandburns** [comment]
LGTM.

**Comment by quinton-hoole** [comment]
Fixes #10622

File: test/e2e/util.go [hunk 1]
```diff
@@ -1466,15 +1467,26 @@ func waitForNodeToBe(c *client.Client, name string, wantReady bool, timeout time
...
+	Logf("Waiting up to %v for all nodes to be ready", timeout)
+
 	var notReady []api.Node
-	for _, node := range nodes.Items {
-		if isNodeReadySetAsExpected(&node, false) {
-			notReady = append(notReady, node)
+	err := wait.Poll(poll, timeout, func() (bool, error) {
+		notReady = nil
+		nodes, err := c.Nodes().List(labels.Everything(), fields.Everything())
+		Expect(err).NotTo(HaveOccurred())
```
Comment (quinton-hoole) [inline test/e2e/util.go]: Isn't it better to just retry?  By omitting this, and instead returning false,err I think you'll be more resilient.

File: test/e2e/util.go [hunk 2]
```diff
@@ -1466,15 +1467,26 @@ func waitForNodeToBe(c *client.Client, name string, wantReady bool, timeout time
...
+		for _, node := range nodes.Items {
+			if !isNodeReadySetAsExpected(&node, true) {
+				notReady = append(notReady, node)
+			}
 		}
+		return len(notReady) == 0, nil
+	})
+
+	if err != nil && err != wait.ErrWaitTimeout {
```
Comment (quinton-hoole) [inline test/e2e/util.go]: Surely you want to return an error on ErrWaitTimeout here?

File: test/e2e/util.go [hunk 3]
```diff
@@ -1432,8 +1432,9 @@ func waitForNodeToBeNotReady(c *client.Client, name string, timeout time.Duratio
 func isNodeReadySetAsExpected(node *api.Node, wantReady bool) bool {
 	// Check the node readiness condition (logging all).
 	for i, cond := range node.Status.Conditions {
-		Logf("Node %s condition %d/%d: type: %v, status: %v",
-			node.Name, i+1, len(node.Status.Conditions), cond.Type, cond.Status)
+		Logf("Node %s condition %d/%d: type: %v, status: %v, reason: %q, message: %q, last transistion time: %v",
+			node.Name, i+1, len(node.Status.Conditions), cond.Type, cond.Status,
+			cond.Reason, cond.Message, cond.LastTransitionTime)
```
Comment (quinton-hoole) [inline test/e2e/util.go]: Not your code, but the if statement below could be made a lot more understandable with s a few brackets, rather relying on implicit operator precedence.

**Comment by quinton-hoole** [comment]
LGTM barring comments.  Let me know when addressed, and we can mark ok-to-merge.

**Comment by quinton-hoole** [comment]
Oh, and we'll want a clean Jenkins e2e test run before merging.

**Comment by krousey** [comment]
@quinton-hoole 

The e2e did pass, but it failed to upload the test artifacts. heh...

> SUCCESS! -- 54 Passed | 0 Failed | 2 Pending | 41 Skipped

File: test/e2e/util.go (hunk 2 above)
Comment (krousey) [inline test/e2e/util.go]: In case of timeout, I want the error message below it to return.

**Comment by krousey** [comment]
@k8s-bot test this please

Another timeout

**Comment by krousey** [comment]
@quinton-hoole Please take another look

File: test/e2e/util.go (hunk 2 above)
Comment (quinton-hoole) [inline test/e2e/util.go]: Doh!  Gotcha.  LGTM.

**Comment by quinton-hoole** [comment]
LGTM.  ok-to-merge.  
This is exclusively test code.

**Comment by krousey** [comment]
Flake averted on job 3124 on e2e-gce-parallel:

```
INFO: Waiting up to 1m0s for all nodes to be ready
INFO: Node e2e-test-parallel-minion-iqg7 condition 1/1: type: Ready, status: False, reason: "network not configured correctly", message: "", last transistion time: 2015-07-09 16:03:37 -0700 PDT
... snip ...
INFO: Node e2e-test-parallel-minion-iqg7 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transistion time: 2015-07-09 16:04:08 -0700 PDT
INFO: Successfully found node e2e-test-parallel-minion-iqg7 readiness to be true
```

**Pull request merged**